Anarchist Library.js

```diff
@@ -0,0 +1,452 @@
+	}
+	return items;
+}
+
+async function scrape(doc, url = doc.location.href) {
+	// ToDo: get fancier here, allow other types
+	let item = new Zotero.Item('webpage');
+
+	// These may be expanded on in the future
+	let attachmentTypes = {
+		PDF: allAttachmentTypes["Plain PDF"],
+	};
+
+	item.accessed = new Date().toString();
+	item.url = url;
+	item.language = attr(doc, "html", "lang");
+
+	let itemType = attr(doc, '[property~="og:type"]', 'content');
+	let tagNodeList = doc.querySelectorAll(`[property~="og:${itemType}:tag"]`);
+	let description = attr(doc, '[property~="og:description"]', 'content');
+	let author = attr(doc, `[property~="og:${itemType}:author"]`, 'content');
+	item.creators.push(ZU.cleanAuthor(author, "author"));
+
+	if (description) {
+		item.description = description;
+		// misses https://theanarchistlibrary.org/library/leo-tolstoy-the-complete-works-of-count-tolstoy-volume-12
+		let re = /(?<=[Tt]ranslated(?: +to [Ee]nglish)? +by ).*$/u;
+		let translatedMatch = description.match(re);
+		if (translatedMatch) {
+			item.creators.push(ZU.cleanAuthor(translatedMatch[0], "translator", translatedMatch[0].includes(",")));
+		}
+	}
+
+	let date = getPreambleVal(doc, "textdate");
+	let notes = getPreambleVal(doc, "preamblenotes");
+	// misses link here: https://theanarchistlibrary.org/library/margaret-killjoy-it-s-time-to-build-resilient-communities
+	let source = getPreambleVal(doc, "preamblesrc");
+
+	for (let tagNode of tagNodeList) {
+		item.tags.push({ tag: tagNode.content });
+	}
+
+	let title = attr(doc.head, '[property~="og:title"][content]', 'content');
+	item.title = title;
+	item.date = date;
+	if (notes) {
+		item.notes.push({ note: ZU.trimInternal(notes) });
+	}
+	if (source) {
+		item.notes.push({ note: `Source: ${ZU.trimInternal(source)}` });
+	}
+
+	for (const [typeName, typeInfo] of Object.entries(attachmentTypes)) {
+		let attachment = {
+			title: typeName,
+			url: `${doc.location.href}${typeInfo.ext}`,
+			mimeType: typeInfo.mimeType
+		};
+
+		if (typeInfo.ext == "snapshot") {
+			attachment.snapshot = true;
+		}
+
+		item.attachments.push(attachment);
+	}
+
+	return item.complete();
+}
+
+var libraryRe = new RegExp(String.raw`library/`);
+
+
+function detectWeb(doc, url) {
+	if (url.match(libraryRe)) {
+		return 'webpage';
+	}
+	else if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+function getPreambleVal(doc, id) {
+	let preamble = doc.body.querySelector("div#preamble");
+	return text(preamble, `div#${id}`).slice(text(preamble, `span#${id}-label`).length);
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+/*
+var unusedTestCases = [
+	// fails tests because Voltairine de Cleyre parses to Cleyre, Voltairine de
```

Comment:
We can remove these; re-add if we get a better name splitting routine in the future. It's always best if sites expose author names in `Last, First` format so it's unambiguous. If they don't... it is what it is. `cleanAuthor()` is fairly naive right now, but the assumption that only the last particle is the family name holds for like 95% of western names, so until we add some better heuristics for name splitting, it's the best we have. If we assumed that only the first particle is the first name, we'd be correct less often.